lib/modules.nix

```diff
@@ -256,6 +256,7 @@ rec {
 
       merged =
         let collected = collectModules
+          specialArgs.class or null
```

Comment:
I don't see a good reason why this should be a `specialArgs` (which gives modules access to that value) instead of just another argument to `evalModules` (and then also `submoduleWith`).

Reply:
It is cheap for the general case of having no class; most submodules don't.
Having this in the module arguments allows modules to adapt to the class they're loaded into, which seems useful.

Reply:
I don't see why it would be expensive to have a separate argument for this.

And using `specialArgs` for this seems wrong to me, they're "special" args because they're just like custom arguments, but more infinite-recursion safe. This change would make `.class` have special semantics relevant for the module system evaluation, which breaks that pattern. It could also break user code fwiw.

> which seems useful.

I'd rather we not expose this to modules at first, unless we have a clear need for it. And if we do have a clear need for it, I'd expose it as `config._module.class` instead, because that definitely won't break user code.

Reply:
Done.